packages/melos/lib/src/commands/run.dart

```diff
@@ -239,15 +239,44 @@ mixin _RunMixin on _Melos {
     await _executeScriptSteps(steps, scripts, script, environment);
   }
 
+  /// Checks if the given [step] is a recognized Melos command.
+  bool _isStepACommand(String step) {
+    // Split the step by spaces to separate the command from its flags/arguments.
+    final command = step.split(' ')[0];
+
+    const melosCommands = {
+      'analyze',
+      'format',
+      'bs',
+      'bootstrap',
+      'clean',
+      'list',
+      'publish',
+    };
+
+    return melosCommands.contains(command);
+  }
+
+  String _buildScriptCommand(String step, Scripts scripts) {
+    if (_isStepACommand(step)) {
+      return 'melos $step';
+    }
+
+    if (scripts.containsKey(step)) {
+      return 'melos run $step';
+    }
```

Comment:
Should these ifs be in the opposite order maybe? So if the user has accidentally defined a script with the same name as a command, it would still work?

Reply:
I introduced this suggestion in commit 0ca796941c449500bdbc45000a41a838605aa9fd and also added a test for this new scenario. :)